storage_controller/src/service.rs

```diff
@@ -5355,9 +5445,29 @@ impl Service {
             waiters = self
                 .await_waiters_remainder(waiters, SHORT_RECONCILE_TIMEOUT)
                 .await;
+
+            failpoint_support::sleep_millis_async!("sleepy-drain-loop");
         }
 
         while !waiters.is_empty() {
+            if cancel.is_cancelled() {
+                match self
+                    .node_configure(node_id, None, Some(NodeSchedulingPolicy::Active))
```

Comment:
This is a bit duplicative: maybe this function should return a cancellation error variant, and on cancellation the outer function can do the node_configure back to the active state?

Reply:
Yeah. I did this way because I wanted to keep the node configuration within the drain/fill functions, but I don't feel strongly.

Reply:
Assuming you didn't feel strongly since you approved in the first place. Resolving.